File: notebooks/npz2r.R
# Use reticulate to convert imdb npz file to Rds.
# Run npz2r.pynb first to download/process imdb data

#devtools::install_github("rstudio/reticulate")

library(reticulate)

## find python binary (run from within python session):
# import sys
# sys.executable

use_python('/home/anta/anaconda3/bin/python')

source_python("npz2r.py")
x_train <- get_data("x_train")
y_train <- get_data("y_train")
x_test <- get_data("x_test")
y_test <- get_data("y_test")

y_train <- as.integer(y_train)
y_test <- as.integer(y_test)

imdb <- list(x_train=x_train, y_train=y_train, x_test=x_test, y_test=y_test)

saveRDS(imdb, file='imdb.Rds')

File: notebooks/npz2r.py
# Function to load imdb.npz. Called by npz2r.R to convert npz to Rds

import numpy

def get_data(dataset):
  with numpy.load('imdb_processed.npz') as f:
    if dataset == "x_train":
      x_train, y_train = f['x_train'], f['y_train']
      return x_train
    elif dataset == "y_train":
      x_train, y_train = f['x_train'], f['y_train']
      return y_train
    elif dataset == "x_test":
      x_test, y_test = f['x_test'], f['y_test']
      return x_test
    elif dataset == "y_test":
      x_test, y_test = f['x_test'], f['y_test']
      return y_test
